modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeDropTableCheck.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.tools.ToolsUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Alan Huang
+ */
+public class JavaUpgradeDropTableCheck extends BaseJavaTermCheck {
+
+	@Override
+	public boolean isLiferaySourceCheck() {
+		return true;
+	}
+
+	@Override
+	protected String doProcess(
+		String fileName, String absolutePath, JavaTerm javaTerm,
+		String fileContent) {
+
+		JavaClass javaClass = (JavaClass)javaTerm;
+
+		List<String> extendedClassNames = javaClass.getExtendedClassNames();
+
+		if (!extendedClassNames.contains("UpgradeProcess")) {
+			return javaTerm.getContent();
+		}
+
+		String content = javaTerm.getContent();
+
+		Matcher matcher = _runSqlPattern.matcher(content);
+
+		while (matcher.find()) {
+			String runSqlMethodCall = _getMethodCall(content, matcher.start());
+
+			if (!runSqlMethodCall.contains("drop table if exists")) {
+				continue;
```

Comment:
Hey @ling-alan-huang, I think the logic is problematic here. Because of the `return` statement in the while loop, the check will only replace **one** occurrence of `drop table if exists` queries.